src/Symfony/Component/Process/Tests/ProcessTest.php

```diff
@@ -1427,9 +1428,10 @@ public function testInheritEnvDisabled()
      *
      * @return Process
      */
-    private function getProcess($commandline, $cwd = null, array $env = null, $input = null, $timeout = 60, array $options = array())
+    private function getProcess($commandline, $cwd = null, array $env = null, $input = null, $timeout = 60)
     {
-        $process = new Process($commandline, $cwd, $env, $input, $timeout, $options);
+        $process = new Process($commandline, $cwd, $env, $input, $timeout);
+        $process->inheritEnvironmentVariables(true, false);
```

Comment:
yes, then no :)

here is the upgrade path:

for people that set $env but do not enable inheriting, they'll see this:

> Not inheriting environment variables is deprecated since Symfony 3.3 and will always happen in 4.0. Call "Process::inheritEnvironmentVariables()" instead before running the command.

so they'll do it.

then people set $env and do enable inheriting, will see this:

> The Process::inheritEnvironmentVariables method is deprecated since version 3.3 and will be removed in 4.0. Environment variables will always be inherited. To silence this notice, set the second argument of "Process::inheritEnvironmentVariables()" to "false", then wrap the call in a "method_exists()" check.

*then wrap the call in a "method_exists()" check.*

see eg [this line](https://github.com/symfony/symfony/pull/21347/files#diff-3dc82e6e990428b0c71cf2112d02269fL278) for the Dotenv component.

I think that's enough an upgrade path for that matter. Better for you with these explanations?